src/GeniusYield/Providers/CardanoDbSync.hs

```diff
@@ -91,6 +94,25 @@ newtype CardanoDbSyncException = CardanoDbSyncException String
   deriving stock (Show)
   deriving anyclass (Exception)
 
+-------------------------------------------------------------------------------
+-- Await tx confirmation
+-------------------------------------------------------------------------------
+
+-- | Awaits for the confirmation of a given 'GYTxId'
+dbSyncAwaitTxConfirmed :: CardanoDbSyncConn -> GYAwaitTx
+dbSyncAwaitTxConfirmed (Conn pool) p@GYAwaitTxParameters{..} txId =
+    Pool.withResource pool $ dbSyncAwaitTx 0
+  where
+    dbSyncAwaitTx :: Int -> PQ.Connection -> IO ()
+    dbSyncAwaitTx attempt _ | maxAttempts <= attempt = throwIO $ GYAwaitTxException p
+    dbSyncAwaitTx attempt conn = do
+        res <- PQ.query conn "SELECT (mtable.max - block_no) as confirmations FROM block, ( SELECT MAX(block_no) as max FROM block ) as mtable, ( SELECT block_id, hash FROM tx WHERE hash = ? ) AS b WHERE block.id = b.block_id;" (PQ.Only txId)
+        case res of
+            [PQ.Only (blockConfirmations :: Int)] ->
+                when (blockConfirmations < confirmations) $
+                threadDelay checkInterval >> dbSyncAwaitTx (attempt + 1) conn
+            _anyOtherMatch -> throwIO $ CardanoDbSyncException "dbSyncAwaitTxConfirmed: zero or multiple SQL results"
```

Comment:
Is the case that the tx is not yet seen on chain handled here? I mean one could submit the tx but it might take time for it to get actually submitted & picked up here. Basically, don't we need to do `threadDelay checkInterval >> dbSyncAwaitTx (attempt + 1) conn` when list is empty?

Reply:
Oh! you are totally right. Big miss, I will fix it :)